addons/sale/models/sale_order.py

```diff
@@ -788,6 +788,14 @@ def _unlink_except_draft_or_cancel(self):
                     "You can not delete a sent quotation or a confirmed sales order."
                     " You must first cancel it."))
 
+    def write(self, vals):
+        for order in self:
+            old_partner_id = order.partner_id.id
+            new_partner_id = vals.get('partner_id', old_partner_id)
+            if order.state in ('sent', 'sale') and old_partner_id != new_partner_id:
+                order.message_subscribe(partner_ids=[new_partner_id])
```

Comment:
I think this is more the Odoo way of doing.
```suggestion
                order.message_subscribe(partner_ids=new_partner_id.ids)
```

Reply:
As discussed, new_partner_id comes from vals, which doesn't contain recordsets, so we'll leave it as is.